src/tests/Makefile

```diff
@@ -3,33 +3,41 @@
 ALL_TEST_CXXFLAGS=$(CPPFLAGS) $(TEST_CXXFLAGS) $(TEST_CXXFLAGS_EXTRA)
 ALL_TEST_LDFLAGS=$(TEST_LDFLAGS) $(TEST_LDFLAGS_EXTRA)
 
-objects = tests.o test-dinit.o proctests.o loadtests.o envtests.o test-run-child-proc.o test-bpsys.o
+objects = tests.o test-dinit.o proctests.o loadtests.o envtests.o iostreamtests.o test-run-child-proc.o test-bpsys.o
 parent_objs = service.o proc-service.o dinit-log.o load-service.o baseproc-service.o dinit-env.o control.o
 
 check: build-tests run-tests
 
-build-tests: tests proctests loadtests envtests
+build-tests: tests proctests loadtests envtests iostreamtests
 	$(MAKE) -C cptests build-tests
 
 run-tests: build-tests
 	./tests
 	./proctests
 	./loadtests
 	./envtests
+	./iostreamtests
 	$(MAKE) -C cptests run-tests
 
 tests: $(parent_objs) tests.o test-dinit.o test-bpsys.o test-run-child-proc.o
 	$(CXX) -o tests $(parent_objs) tests.o test-dinit.o test-bpsys.o test-run-child-proc.o $(ALL_TEST_LDFLAGS)
 
 proctests: $(parent_objs) proctests.o test-dinit.o test-bpsys.o test-run-child-proc.o
 	$(CXX) -o proctests $(parent_objs) proctests.o test-dinit.o test-bpsys.o test-run-child-proc.o $(ALL_TEST_LDFLAGS)
-	
+
 loadtests: $(parent_objs) loadtests.o test-dinit.o test-bpsys.o test-run-child-proc.o
 	$(CXX) -o loadtests $(parent_objs) loadtests.o test-dinit.o test-bpsys.o test-run-child-proc.o $(ALL_TEST_LDFLAGS)
 
 envtests: $(parent_objs) envtests.o test-dinit.o test-bpsys.o test-run-child-proc.o
 	$(CXX) -o envtests $(parent_objs) envtests.o test-dinit.o test-bpsys.o test-run-child-proc.o $(ALL_TEST_LDFLAGS)
 
+# This is separate target than parent_objs because of some issues of closing default fds in base-bpsys.cc
```

Comment:
This is a bit vague.

Something like:

    # dinit-iostream contains global objects which require special handling in test environments.
    # Therefore, avoid linking it to tests by default (instead include it explicitly in relevant tests eg iostreamtests).

... would be more useful I think.